packages/cadl-ranch-specs/http/models/inheritance/main.cadl

```diff
@@ -1,60 +1,112 @@
 import "@azure-tools/cadl-ranch-expect";
 
 using Cadl.Rest;
 using Cadl.Http;
 using Azure.Core.Foundations;
 
-@doc("Illustrates round-trip polymorphic models.")
-@scenarioService("/polymorphic")
-namespace BasicPolymorphicModels;
+@serviceTitle("ComplexInheritanceClient")
+@serviceVersion("1.0.0")
+@server("http://localhost:3000", "Testserver endpoint")
+@doc("Illustrates inheritance model.")
+@route("/inheritance")
+namespace Model.Inheritance;
 
-@doc("Example base type.")
-model BaseClass {
-  @doc("An example property.")
-  baseClassProperty: string;
+@doc("Base model for non-discriminator three levels inheritance.")
+model Pet {
+  name?: string;
 }
 
-model DerivedFromBaseClassA extends BaseClass {
-  @doc("An example property on a derived type")
-  derivedClassAProperty: string;
+@doc("The second level model A in the non-discriminator three levels inheritance.")
+model Cat extends Pet {
+  age?: int32;
 }
 
-model DerivedFromBaseClassB extends BaseClass {
-  @doc("An example property on a derived type")
-  derivedClassBProperty: string;
+@doc("The third level model A in the non-discriminator three levels inheritance.")
+model Siamese extends Cat {
+  smart?: boolean;
 }
 
-@discriminator("discriminatorProperty")
-@doc("Example base class that has a discriminator property.")
-model BaseClassWithDiscriminator extends BaseClass {}
-// Note: discriminator property is not defined on parent
+@doc("Base model for three levels inheritance with discriminator.")
+@discriminator("kind")
+model Fish {
+  age?: int32;
+}
+
+@doc("The second level model in three levels inheritance with discriminator and defined a new discriminator.")
+@discriminator("sharktype")
+model Shark extends Fish {
+  kind: "shark";
+}
 
-model DerivedFromBaseClassWithDiscriminatorA extends BaseClassWithDiscriminator {
-  discriminatorProperty: "A";
+@doc("The second level model in three levels inheritance with discriminator.")
+model Salmon extends Fish {
+  kind: "salmon";
+  friends?: Fish[];
+  hate?: Record<Fish>;
+  partner?: Fish;
 }
 
-model DerivedFromBaseClassWithDiscriminatorB extends BaseClassWithDiscriminator {
-  discriminatorProperty: "B";
+@doc("The third level model A in the three levels inheritance with discriminators.")
+model SawShark extends Shark {
+  sharktype: "saw";
 }
 
-@doc("Illustrates case where a basic model has polymorphic properties.")
-model ModelWithPolymorphicProperty {
-  @doc("Example polymorphic type property.")
-  polymorphicProperty: BaseClassWithDiscriminator;
+@doc("The third level model B in the non-discriminator three levels inheritance with discriminators.")
+model GoblinShark extends Shark {
+  sharktype: "goblin";
 }
 
 @scenario
-@scenarioDoc("Generate, send, and receive round-trip inherited model.")
-@route("/model")
+@scenarioDoc("Generate and send model. The valid input value is {name: \"abc\", age: 32, smart: true}")
+@route("/valid")
+@post
+op sendInheritanceValid(@body input: Siamese): OkResponse | ErrorResponse;
```

Comment:
nit: you don't need to repeat `inheritance` in the name, you can just say `putValid` and `getValid` etc

Reply:
done